File: blastamon.java
/****************************************************************************
 *          17th Annual UCF High School Programming Tournament              *
 *                              May 2, 2003                                 *
 *                     University of Central Florida                        *
 *                                                                          *
 *                                                                          *
 * Special thanks to our sponsors: UPE, SAIC, ACM, and HARRIS               *
 *                                                                          *
 *                                                                          *
 * Problem:         Blastamon                                               *
 * Problem Author:  Jason                                                   *
 * Solution Author: Mike W                                                  *
 * Data Author:     Nick                                                    *
 ***************************************************************************/
import java.io.*;

public class blastamon
{

  String[] values;	
  BufferedReader in;

  public void initialize()
  {
    //Makes sure that all the strings stored
    //are set to null

    values = new String[1001];
    for(int i=0;i<1001;i++)
      values[i]=null;
  }

  public void process(int numberOfEntries)
  {
    int placeholder=0;
    String readName=null;

    initialize();
		
    try{
      for(int i=0;i<numberOfEntries;i++)
      {
        //Read in the name and hold on to it
        readName=in.readLine();

        //read in the hit points (place value)
        placeholder = Integer.parseInt(in.readLine());

        //Store it
        values[placeholder]=readName;
      }
    }
    catch(IOException e){}
  }

  public void output()
  {

    //The Entries are already sorted by hit points now
    //because of the way we stored them
    //No ties means no two creatures will have the same hitpoints
    //That way no creature will overwrite another in the list of names
    //So print them starting from the beginning.

    for(int i=0;i<1001;i++)
    {
      if(values[i]!=null)
        System.out.println(values[i]);
    }
  }
	
  public boolean running()
  {
    int numberOfEntries=0;

    //This part reads in the number of entries for this dictionary

    try{
      numberOfEntries = Integer.parseInt(in.readLine());
    }catch(IOException e){}


    if(numberOfEntries==0) return false;

    process(numberOfEntries);		
    output();

    return true;
  }

  public void run()
  {


    in=null;
	
    try{
      in = new BufferedReader(new FileReader("blastamon.in"));
    }catch(IOException e){}


    //continue processing till the specified condition is met.
    while(running()) System.out.println();
  }

  public static void main(String[] args)
  {
    blastamon b = new blastamon();
    b.run();
  }
}

File: blastamon.cpp
/****************************************************************************
 *          17th Annual UCF High School Programming Tournament              *
 *                              May 2, 2003                                 *
 *                     University of Central Florida                        *
 *                                                                          *
 *                                                                          *
 * Special thanks to our sponsors: UPE, SAIC, ACM, and HARRIS               *
 *                                                                          *
 *                                                                          *
 * Problem:         Blastamon                                               *
 * Problem Author:  Jason                                                   *
 * Solution Author: Mike W                                                  *
 * Data Author:     Nick                                                    *
 ***************************************************************************/
#include<stdio.h>
#include<string.h>
#include<stdlib.h>

FILE * in;
char values[1001][21];

void initialize()
{
  int i;

  for(i=0;i<1001;i++)
  {
    //in c '\0' is the end of string marker
    //By setting the first index of the string
    // to '\0' I'm officialy setting the string to ""
    values[i][0]='\0';
  }
}

void process(int numberOfEntries)
{
  int i;
  int temp;
  char tempstr[21];

  initialize();

  for(i=0;i<numberOfEntries;i++)
  {

    //Store in this particular entry
    fscanf(in,"%s\n",tempstr);

    fscanf(in,"%d\n",&temp);


    //Copy it to its place in the list
    strcpy(values[temp],tempstr);
  }


}

void printList()
{
  int i;
  for(i=0;i<1001;i++)
  {
    //Print anything that isn't ""
    if(values[i][0]!='\0')
      printf("%s\n",values[i]);
  }
}

int running()
{
  int numberOfEntries;

  //Get the number of Entries for this case
  fscanf(in,"%d\n",&numberOfEntries);

  if(numberOfEntries==0) return 0;


  //Process them and then print them
  process(numberOfEntries);

  printList();

  return 1;
}

int main()
{
  in = fopen("blastamon.in","r");

  //Keep processing the input till specific case is found
  while(running()) printf("\n");	

  fclose(in);

  return 0;
}
